### PR TITLE
User get notified of permission errors on the frontend

### PR DESCRIPTION
**Pull request by gsandok**
Message has format of "The user [example@example.com](mailto:example@example.com) with role <role> is not allow to perform the action <action> on environment <env> for team <team>"

Error toast looks like this: 
Screenshot 2023-09-07 at 7 12 00 PM

### CODE REVIEW

**Review by hannesg — APPROVED** [review]

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx [hunk 1]
```diff
@@ -343,7 +343,11 @@ export const SideBar: React.FC<{ className?: string; toggleSidebar: () => void }
                 .catch((e) => {
                     // eslint-disable-next-line no-console
                     console.error('error in batch request: ', e);
-                    showSnackbarError('Actions were not applied. Please try again');
+                    if (e.message.includes('not allowed to perform the action')) {
```
[the remark this answers is not in the document]
Comment (gsandok) [reply services/frontend-service/src/ui/components/SideBar/SideBar.tsx:346]: I would be fine always showing the error message and can update that now

File: pkg/auth/rbac.go [hunk 2]
```diff
@@ -221,8 +221,7 @@ func CheckUserPermissions(rbacConfig RBACConfig, user *User, env, envGroup, appl
 		}
 	}
 	// The permission is not found. Return an error.
-	p := fmt.Sprintf(PermissionTemplate, user.DexAuthContext.Role, action, envGroup, env, application)
-	return status.Errorf(codes.PermissionDenied, fmt.Sprintf("user does not have permissions for: %s", p))
+	return status.Errorf(codes.PermissionDenied, fmt.Sprintf("The user %s with role %s is not allowed to perform the action %s on environment %s for team %s", user.Name, user.DexAuthContext.Role, action, env, team))
```
Comment (sven-urbanski-freiheit-com) [inline pkg/auth/rbac.go:224]: Please add some quotes:
```suggestion
	return status.Errorf(codes.PermissionDenied, fmt.Sprintf("The user '%s' with role '%s' is not allowed to perform the action '%s' on environment '%s' for team '%s'", user.Name, user.DexAuthContext.Role, action, env, team))
```

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/SideBar/SideBar.tsx:346]: I would prefer to check by status code.
If it's authentication related -> Show your message.
Else: Show generic message.

If there's an "internal server error", the message would probably not be very user friendly, so I'd rather have the "try again" message instead.

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
see comment about status code

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx [hunk 3]
```diff
@@ -343,7 +343,11 @@ export const SideBar: React.FC<{ className?: string; toggleSidebar: () => void }
                 .catch((e) => {
                     // eslint-disable-next-line no-console
                     console.error('error in batch request: ', e);
-                    showSnackbarError('Actions were not applied. Please try again');
+                    if (e.message.includes('PermissionDenied')) {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/SideBar/SideBar.tsx:346]:
```suggestion
                    const GrpcErrorPermissionDenied = 7;
                    if (e.code === GrpcErrorPermissionDenied) {
```

Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/SideBar/SideBar.tsx:346]: please do not rely on the text in the message. That's what the code is made for.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**